catalog/rocks/staging/iterator.go

```diff
@@ -0,0 +1,68 @@
+package staging
+
+import (
+	"context"
+
+	"github.com/treeverse/lakefs/catalog/rocks"
+)
+
+const batchSize = 1000
+
+type Iterator struct {
+	mgr         *Manager
+	nextFrom    rocks.Path
+	idxInBuffer int
+	err         error
+	ctx         context.Context
+	st          rocks.StagingToken
+	dbHasNext   bool
+	buffer      []*rocks.EntryRecord
+}
+
+func NewIterator(ctx context.Context, mgr *Manager, st rocks.StagingToken, nextFrom rocks.Path) *Iterator {
```

Comment:
just something I thought about - with this implementation we can simplify the creation and remove the 'nextFrom'.
it is just a state change we can perform by calling to SeekGE after NewIterator.